asan_fake_stack.cpp:78:11: warning: format ‘%zx’ expects argument of type ‘size_t’,

https://github.com/llvm/llvm-project/blob/8ab6b38dab63fda49e6294e3b0d847ae491d7701/compiler-rt/lib/asan/asan_fake_stack.cpp#L77-L84

Compiling GCC I see:

`gcc/libsanitizer/asan/asan_fake_stack.cpp:78:11: warning: format ‘%zx’ expects argument of type ‘size_t’, but argument 9 has type ‘__sanitizer::u8*’ {aka ‘unsigned char*’} [-Wformat=]`

I don't know whether it should be `%p` and the argument should be cast to `(void*)` or if `0x%zx` is correct and the argument should be cast to `size_t` but it should be fixed.

--- a/compiler-rt/lib/asan/asan_fake_stack.cpp
+++ b/compiler-rt/lib/asan/asan_fake_stack.cpp
@@ -77,11 +77,11 @@ FakeStack* FakeStack::Create(uptr stack_size_log) {
   VReport(1,
           "T%d: FakeStack created: %p -- %p stack_size_log: %zd; "
           "mmapped %zdK, noreserve=%d, true_start: %p, start of first frame: "
-          "0x%zx\n",
+          "%p\n",
           GetCurrentTidOrInvalid(), (void*)p,
           (void*)(p + FakeStack::RequiredSize(stack_size_log)), stack_size_log,
           size >> 10, flags()->uar_noreserve, res->true_start,
-          res->GetFrame(stack_size_log, /*class_id*/ 0, /*pos*/ 0));
+          (void*)res->GetFrame(stack_size_log, /*class_id*/ 0, /*pos*/ 0));
   return res;
 }
 

